cmake/module/FindUSDT.cmake

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2024-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+#[=======================================================================[
+FindUSDT
+--------
+
+Finds the Userspace, Statically Defined Tracing header(s).
+
+Imported Targets
+^^^^^^^^^^^^^^^^
+
+This module provides imported target ``USDT::headers``, if
+USDT has been found.
+
+Result Variables
+^^^^^^^^^^^^^^^^
+
+This module defines the following variables:
```

Comment:
Why? If it's not exported / used anywhere outside of this.

Reply:
1. It is expected from any Find module.

From CMake [docs](https://cmake.org/cmake/help/latest/manual/cmake-developer.7.html#find-modules):
> The primary task of a find module is to determine whether a package is available, set the `<PackageName>_FOUND` variable to reflect this...

2. It was used in the first commit. However, the second commit removed that usage.

Reply:
> However, the second commit removed that usage.

So then it can be removed? Or are we adding dead/unused code to our build system because of what the CMake docs say? Or are we doing things incorrectly, if we aren't using the variables we are adding, according to the docs?